extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetMetadataFileWriterImpl.java

```diff
@@ -253,21 +250,18 @@ private static void mergeBlocksInto(final ParquetFileMetadata parquetFileMetadat
         }
     }
 
-    private static String getRelativePath(final String parquetFilePath, final Path metadataRootDirAbsPath) {
-        final Path parquetFileAbsPath = new File(parquetFilePath).getAbsoluteFile().toPath();
-        return metadataRootDirAbsPath.relativize(parquetFileAbsPath).toString();
-    }
-
-    private void writeMetadataFile(final ParquetMetadata metadataFooter, final String outputPath) throws IOException {
-        final PositionedBufferedOutputStream metadataOutputStream =
-                new PositionedBufferedOutputStream(channelsProvider.getWriteChannel(outputPath, false),
-                        ParquetUtils.PARQUET_OUTPUT_BUFFER_SIZE);
-        metadataOutputStream.write(MAGIC);
-        ParquetFileWriter.serializeFooter(metadataFooter, metadataOutputStream);
-        metadataOutputStream.close();
+    private static void writeMetadataFile(final ParquetMetadata metadataFooter, final OutputStream outputStream)
+            throws IOException {
+        final CountingOutputStream countingOutputStream = new CountingOutputStream(outputStream);
+        countingOutputStream.write(MAGIC);
+        ParquetFileWriter.serializeFooter(metadataFooter, countingOutputStream);
+        countingOutputStream.flush();
```

Comment:
Ditto; if owner is responsible, I don't see too much value in flushing.

Reply:
Same as above.